### PR TITLE
separate out static dependencies of core on java backend

### PR DESCRIPTION
**Pull request by dwightguth**
@andreistefanescu @yilongli @cos please review

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by yilongli** [comment]
lgtm

File: src/javasources/KTool/src/org/kframework/kompile/KompileOptions.java [hunk 1]
```diff
@@ -123,16 +44,20 @@ public File mainDefinitionFile() {
...
+//    LATEX(true, false, LatexBackend.class, null, "autoinclude.k"),
+//    DOC(true, false, DocumentationBackend.class, null, "autoinclude.k"),
+//    HTML(true, false, HtmlBackend.class, null, "autoinclude.k"),
+//    MAUDE(false, true, KompileBackend.class, MaudeKRun.class, "autoinclude.k"),
+//    JAVA(false, true, JavaSymbolicBackend.class, JavaSymbolicKRun.class, "autoinclude-java.k"),
+//    UNPARSE(false, false, UnparserBackend.class, null, "autoinclude.k"),
+//    UNFLATTEN(false, false, UnflattenBackend.class, null, "autoinclude.k"),
+//    UNFLATTEN_JAVA(false, false, UnflattenJavaBackend.class, null, "autoinclude-java.k"),
+//    SYMBOLIC(false, true, SymbolicBackend.class, MaudeKRun.class, "autoinclude.k"),
+//    COQ(false, false, CoqBackend.class, null, "autoinclude-java.k");
```
Comment (cos) [inline src/javasources/KTool/src/org/kframework/kompile/KompileOptions.java]: should we remove this comment?

**Comment by cos** [comment]
looks better than before (also, the green is smaller than the red: +557/−621)

File: src/javasources/KTool/src/org/kframework/kompile/KompileOptions.java (hunk 1 above)
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/kompile/KompileOptions.java]: oh yes, thank you, I forgot to do that.

**Pull request merged**